pkg/provider/pulumi/common/image.go

```diff
@@ -75,7 +75,9 @@ func NewImage(ctx *pulumi.Context, name string, args *ImageArgs, opts ...pulumi.
 		return nil, err
 	}
 
-	relDocker, err := filepath.Rel(args.ProjectDir, dockerFilePath)
+	dockerContext := dockerContext(args.ProjectDir, args.Compute)
+
+	relDocker, err := filepath.Rel(dockerContext, dockerFilePath)
```

Comment:
These pulumi builds will be deprecated in the near future, and changes related to their context probably should not be implemented (the resulting docker image doesn't/shouldn't actually copy files as it should only be wrapping an existing docker source image.